packages/path_provider/path_provider_android/test/path_provider_android_test.dart

```diff
@@ -89,13 +89,9 @@ void main() {
       });
     } // end of for-loop
 
-    test('getDownloadsPath fails', () async {
-      try {
-        await pathProvider.getDownloadsPath();
-        fail('should throw UnsupportedError');
-      } catch (e) {
-        expect(e, isUnsupportedError);
-      }
+    test('getDownloadsPath succeeds', () async {
+      final String? path = await pathProvider.getDownloadsPath();
+      expect(path, kExternalStoragePaths);
```

Comment:
There are two code paths through the new method, success and failure. Each path needs a unit test.

Reply:
i've updated the new method to return a nullable path

Reply:
The current function now only returns a path, one code path, so I think 1 test is enough? (getDownloadsPath succeed)

Cause it no longer throws an error, hence in test we can't expect it to throw something.

Reply:
> The current function now only returns a path, one code path, so I think 1 test is enough? (getDownloadsPath succeed)

As my comment above says, there is a bug in your current implementation, which is compelling evidence that the current test is insufficient.

> Cause it no longer throws an error

Please test your code with an empty return array if you don't believe me that your code can still throw.